signer/src/transaction_coordinator.rs

```diff
@@ -390,6 +392,95 @@ where
         .map(|_| ())
     }
 
+    /// Constructs a BitcoinPreSignRequest from the given transaction package and
+    /// sends it to the signers. Waits for acknowledgments from the signers until
+    /// the threshold is met or a timeout occurs.
+    /// If the signal stream closes unexpectedly, triggers a shutdown.
+    #[tracing::instrument(skip_all)]
+    async fn construct_and_send_bitcoin_presign_request(
+        &mut self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+        pending_requests: &SbtcRequests,
+        transaction_package: &[utxo::UnsignedTransaction<'_>],
+    ) -> Result<(), Error> {
+        // Create the BitcoinPreSignRequest from the transaction package
+        let sbtc_requests = BitcoinPreSignRequest {
+            request_package: transaction_package
+                .iter()
+                .map(|tx| (&tx.requests).into())
+                .collect(),
+            fee_rate: pending_requests.signer_state.fee_rate,
+            last_fees: pending_requests.signer_state.last_fees.map(Into::into),
+        };
+
+        let presign_ack_filter = |event: &SignerSignal| {
+            matches!(
+                event,
+                SignerSignal::Event(SignerEvent::TxSigner(TxSignerEvent::MessageGenerated(_)))
+                    | SignerSignal::Event(SignerEvent::P2P(P2PEvent::MessageReceived(_)))
+                    | SignerSignal::Command(SignerCommand::Shutdown)
+            )
+        };
+
+        // Create a signal stream with the defined filter
+        let signal_stream = self.context.as_signal_stream(presign_ack_filter);
+
+        // Send the presign request message
+        self.send_message(sbtc_requests, bitcoin_chain_tip).await?;
+
+        tokio::pin!(signal_stream);
+        let future = async {
+            let target_tip = *bitcoin_chain_tip;
+            let mut acknowledged_signers = hashbrown::HashSet::new();
```

Comment:
We should use the standard library `HashSet`, I believe that a lot of the hashbrown implementation has already been ported over there. And I want to get WSTS to use the one in the standard library as well.

Reply:
Yeah, hashbrown is pretty much a part of stdlib now, so if we don't need hashbrown for specific reasons we should use std, agreed

Reply:
done in https://github.com/stacks-network/sbtc/pull/956/commits/c803d9d18bc48fc3aae449ca3c59233a26f44228. Should we open a tech debt issue about removing all of our use of `hashbrown::HashSet`  and `HashMap`? Just to make it clear

Reply:
Maybe, but it's here because WSTS uses it. We cannot unilaterally remove it.